drivers/flash/nrf_qspi_nor.c

```diff
@@ -557,6 +576,49 @@ static int qspi_nrfx_configure(const struct device *dev)
 			}
 		}
 
+		if (ret < 0) {
+			LOG_ERR("QE %s failed: %d", qe_value ? "set" : "clear",
+				ret);
+			return ret;
+		}
+	} else if (INST_0_QER == JESD216_DW15_QER_S2B1v4) {
+
+		ret = qspi_rdsr2(dev);
+		if (ret < 0) {
+			LOG_ERR("RDSR failed: %d", ret);
+			return ret;
+		}
+
+		uint8_t sr = (uint8_t)ret;
+		const uint8_t qe_mask = BIT(1); /* only S2B1 */
+		bool qe_state = ((sr & qe_mask) != 0U);
+
+		LOG_DBG("RDSR %02x QE %d need %d: %s", sr, qe_state, qe_value,
+			(qe_state != qe_value) ? "updating" : "no-change");
+
+		ret = 0;
+		if (qe_state != qe_value) {
```

Comment:
Code for support S2B1V6 is very similar to SB1V4. It Looks like almost copy-pasted.
Can this be rewritten so common code will be used for both?

Reply:
As far as I can tell, S2B1V4 and S2B1V6 are really the same method as far as we're concerned. I'll update this to work for both. I'll also update to address some of the comments on #36567